internal/engine/projected_table_iterator.go

```diff
@@ -0,0 +1,104 @@
+package engine
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/xqueries/xdb/internal/compiler/command"
+	"github.com/xqueries/xdb/internal/engine/table"
+	"github.com/xqueries/xdb/internal/engine/types"
+)
+
+type projectedTableIterator struct {
```

Comment:
Maybe mention what all this can do - allows us to just glance here and know the functions.